k2/csrc/fsa_algo.cu

```diff
@@ -258,4 +302,50 @@ void ArcSort(Fsa &src, Fsa *dest, Array1<int32_t> *arc_map /*= nullptr*/) {
   *dest = tmp;
 }
 
+double ShortestDistance(Fsa &fsa, Array1<int32_t> *arc_map /*= nullptr*/) {
+  ContextPtr &context = fsa.Context();
+  K2_CHECK_EQ(fsa.NumAxes(), 2);
+  K2_CHECK_EQ(context->GetDeviceType(), kCpu);
+  int32_t num_states = fsa.Dim0();
+  k2host::Fsa host_fsa = FsaToHostFsa(fsa);
+  Array1<double> state_weights(context, num_states);
+  std::vector<int32_t> tmp_arc_map;
+  ComputeForwardMaxWeights(host_fsa, state_weights.Data(), &tmp_arc_map);
+  if (arc_map != nullptr) *arc_map = Array1<int32_t>(context, tmp_arc_map);
+
+  return state_weights.Back();
```

Comment:
Calling `Back` on an empty vector will cause undefined behavior.

Reply:
I think the behavior is well defined

https://github.com/k2-fsa/k2/blob/4b50f519e0e6a7de5214b52a32741c1f8cbbdb8c/k2/csrc/array.h#L243-L246

Reply:
We should not die here, we should return negative_infinity. Or we should at least add documentation to say what will happen when it's an empty FSA. (For empty FSA, I believe we should not die)

Reply:
Fixed. Return negative infinity on empty FSAs.